src/sentry/snuba/discover.py

```diff
@@ -136,6 +136,61 @@ def create_reference_event_conditions(reference_event):
     return conditions
 
 
+# TODO (evanh) This whole function is here because we are using the max value to
+# calculate the entire bucket width. If we could do that in a smarter way,
+# we could avoid this whole calculation.
+def find_histogram_buckets(field, params):
+    match = is_function(field)
+    if not match:
+        raise InvalidSearchQuery(u"received {}, expected histogram function".format(field))
+
+    columns = [c.strip() for c in match.group("columns").split(",") if len(c.strip()) > 0]
+
+    if len(columns) != 2:
+        raise InvalidSearchQuery(
+            u"histogram(...) expects 2 column arguments, received {:g} arguments".format(
+                len(columns)
+            )
+        )
+
+    column = columns[0]
+    # TODO evanh: This can be expanded to more fields at a later date, for now keep this limited.
+    if column != "transaction.duration":
+        raise InvalidSearchQuery(
+            "histogram(...) can only be used with the transaction.duration column"
+        )
+
+    try:
+        num_buckets = int(columns[1])
+        if num_buckets < 1 or num_buckets > 500:
+            raise Exception()
+    except Exception:
+        raise InvalidSearchQuery(
+            u"histogram(...) requires a bucket value between 1 and 500, not {}".format(columns[1])
+        )
+
+    alias = u"max_{}".format(column)
+
+    results = raw_query(
+        filter_keys={"project_id": params.get("project_id")},
+        start=params.get("start"),
+        end=params.get("end"),
+        dataset=Dataset.Discover,
+        conditions=[["type", "=", "transaction"]],
+        aggregations=[["max", "duration", alias]],
+    )
+    if "error" in results or len(results["data"]) != 1:
+        raise InvalidSearchQuery("Invalid histogram bucket call")
```

Comment:
Should we capture this error in sentry? It seems like something that shouldn't happen during normal operation.

Reply:
So what I did instead is let it bubble up to the API handler, which will capture the error in sentry.